File: mail-service/src/main/java/com/orderup/service/mail/dto/request/ActivationAttemptRequest.java
package com.orderup.service.mail.dto.request;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class ActivationAttemptRequest {
    private UUID userId;
    private Long otpCode;
    private LocalDateTime requestDate;
    private LocalDateTime expirationDate;
}

File: user-service/bin/main/com/saintrivers/controltower/model/request/ActivationAttemptRequest.kt
package com.saintrivers.controltower.model.request

import com.saintrivers.controltower.model.entity.ActivationAttempt
import java.time.LocalDateTime
import java.util.*

data class ActivationAttemptRequest (
    val userId: UUID,
    val otpCode: Long,
    val requestDate: LocalDateTime,
    val expirationDate:LocalDateTime
    ) {
    fun toEntity() = ActivationAttempt(
        userId = userId,
        otpCode = otpCode,
        requestDate = requestDate,
        expirationDate = expirationDate
    )
}